### ISSUE TITLE
EXCEPTION_ACCESS_VIOLATION

### ISSUE THREAD

**Issue by mosheduminer** [oracle/graal#2954]
**Describe the issue**
Building a native image with graalvm native-image results in `EXCEPTION_ACCESS_VIOLATION` at build time.

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. clone [this repo](https://github.com/mosheduminer/jmonkey-hello-world).
2. **modify the path to the local graal distribution** in `gradle.build`.
3. run `gradlew build shadowjar`.
4. run `native-image -jar ./build/libs/hello_world-0.1-SNAPSHOT-all.jar --allow-incomplete-classpath -H:NativeLinkerOption=prefs.lib --initialize-at-build-time`

**Describe GraalVM and your environment:**
I had this error with 20.2 as well as the latest graal-dev distribution.

 - JDK major version: 11
 - OS: Windows 10
 - Architecture: AMD64

**More details**

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x0000000000000000, pid=10220, tid=14228
#
# JRE version: OpenJDK Runtime Environment GraalVM CE 20.2.0 (11.0.8+10) (build 11.0.8+10-jvmci-20.2-b03)
# Java VM: OpenJDK 64-Bit Server VM GraalVM CE 20.2.0 (11.0.8+10-jvmci-20.2-b03, mixed mode, tiered, jvmci, jvmci compiler, compressed oops, parallel gc, windows-amd64)
# Problematic frame:
# C  0x0000000000000000
#
# No core dump will be written. Minidumps are not enabled by default on client versions of Windows
#
# An error report file with more information is saved as:
# C:\Users\moshe\Desktop\Foolish_Wisdom\kotlin\jmonkey\hello_world\hs_err_pid10220.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/oracle/graal/issues
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
Error: Image build request failed with exit status 1
```

You can find the dump referenced in the repo I linked above. This dump is from running 20.2, I ran the same command with the latest dev release, you can the dump from that run in the `hs_err_pid3816.log` file.

**Comment by dougxc** [comment]
Tracked internally with GR-26774.

**Comment by tkrodriguez** [comment]
Can you attach an hs_err_pid please?

**Comment by mosheduminer** [comment]
It is in the repo linked. I can attach it here when I get back to my computer.

**Comment by tkrodriguez** [comment]
Oh I see.  Somehow it didn't click for me that the logs were in the repo.  I'll grab them from there.

**Comment by tkrodriguez** [comment]
This looks like a native image issue with initializing classes that aren't safe to just statically initialize.  @christianwimmer 
```
j  org.lwjgl.opencl.CL.ngetFunctionAddress(J)J+0
j  org.lwjgl.opencl.CL.getFunctionAddress(Ljava/lang/String;)J+9
j  org.lwjgl.opencl.CLCapabilities.<clinit>()V+2
v  ~StubRoutines::call_stub
J 2855  jdk.internal.misc.Unsafe.ensureClassInitialized0(Ljava/lang/Class;)V java.base@11.0.9 (0 bytes) @ 0x0000022a10a9640f [0x0000022a10a963c0+0x000000000000004f]
J 2956 c1 com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.ensureClassInitialized(Ljava/lang/Class;Z)Lcom/oracle/svm/hosted/classinitialization/InitKind; (160 bytes) @ 0x0000022a09a5bbc4 [0x0000022a09a5ba60+0x0000000000000164]
J 8569 jvmci com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(Ljava/lang/Class;Z)Lcom/oracle/svm/hosted/classinitialization/InitKind; (279 bytes) @ 0x0000022a10da070c [0x0000022a10da0120+0x00000000000005ec]
J 6632 c1 com.oracle.svm.hosted.SVMHost.initializeType(Lcom/oracle/graal/pointsto/meta/AnalysisType;)V (124 bytes) @ 0x0000022a0a73dcfc [0x0000022a0a73cd20+0x0000000000000fdc]
J 7541 c1 com.oracle.graal.pointsto.meta.AnalysisType$$Lambda$200.run()V (12 bytes) @ 0x0000022a0ab06aec [0x0000022a0ab06840+0x00000000000002ac]
```

**Comment by christianwimmer** [comment]
The command line shown in the issue description contains `--initialize-at-build-time`. That is not a good idea in general, but especially not for a library that for sure uses JNI (and therefore likely has `System.loadLibrary` in a class initializer) and is likely to cache C state in `static final` fields.

**Comment by wirthi** [comment]
According to the internal ticket, this has been fixed in GraalVM 21.0. and later
